cmd/prometheus/main.go

```diff
@@ -160,9 +160,15 @@ func main() {
 		"Maximum duration compacted blocks may span. For use in testing. (Defaults to 10% of the retention period).").
 		Hidden().PlaceHolder("<duration>").SetValue(&cfg.tsdb.MaxBlockDuration)
 
-	a.Flag("storage.tsdb.retention", "How long to retain samples in storage.").
+	a.Flag("storage.tsdb.retention", "How long to retain samples in storage. (This flag has been deprecated. Use storage.tsdb.retention.time as a replacement.)").
```

Comment:
```suggestion
	// [DEPRECATED] use storage.tsdb.retention.time instead.
	a.Flag("storage.tsdb.retention", "How long to retain samples in storage. (This flag has been deprecated. Use storage.tsdb.retention.time as a replacement.)").
```

this will allows to do a global search and remove the obsolete code.